src/category_theory/adjunction/fully_faithful.lean

```diff
@@ -63,11 +64,60 @@ instance counit_is_iso_of_R_fully_faithful [full R] [faithful R] : is_iso (adjun
     simp,
   end⟩⟩⟩
 
--- TODO also prove the converses?
--- def L_full_of_unit_is_iso [is_iso (adjunction.unit h)] : full L := sorry
--- def L_faithful_of_unit_is_iso [is_iso (adjunction.unit h)] : faithful L := sorry
--- def R_full_of_counit_is_iso [is_iso (adjunction.counit h)] : full R := sorry
--- def R_faithful_of_counit_is_iso [is_iso (adjunction.counit h)] : faithful R := sorry
+/-- If the unit of an adjunction is an isomorphism, then its inverse on the image of L is given
+by L whiskered with the counit. -/
+@[simp]
+lemma inv_map_unit {X : C} [is_iso (h.unit.app X)] :
+  inv (L.map (h.unit.app X)) = h.counit.app (L.obj X) :=
+is_iso.inv_eq_of_hom_inv_id (h.left_triangle_components)
+
+/-- If the unit of an adjunction is an isomorphism, then its inverse on the image of L is given
+by L whiskered with the counit. -/
+@[simps]
+noncomputable def whisker_left_L_counit_iso_of_is_iso_unit [is_iso (adjunction.unit h)] :
+  L ⋙ R ⋙ L ≅ L :=
+(functor.associator _ _ _).symm ≪≫
+  iso_whisker_right (as_iso (adjunction.unit h)).symm L ≪≫ functor.left_unitor _
+
+/-- If the counit of an adjunction is an isomorphism, then its inverse on the image of R is given
+by R whiskered with the unit. -/
+@[simp]
+lemma inv_counit_map {X : D} [is_iso (h.counit.app X)] :
+  inv (R.map (h.counit.app X)) = (h.unit.app (R.obj X)) :=
```

Comment:
```suggestion
  inv (R.map (h.counit.app X)) = h.unit.app (R.obj X) :=
```
These parens aren't necessary - it does take a bit of time to get used to which parens are needed unless you think through the precedence rules a lot, but an easy thing you can do is just to remove parens and see if it works! Similarly in the next line `is_iso.inv_eq_of_inv_hom_id h.right_triangle_components` works